### PR TITLE
#BAEL-2802

### PR DESCRIPTION
**Pull request by maryarm**
Implement response decoder for ok http client and apply review notes

### CODE REVIEW

**Comment by asif-anwar** [comment]
[retest this please]
Note: This comment is to trigger the Jenkins CI build

File: libraries-2/src/test/java/com/baeldung/okhttp/ResponseDecoderUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+                .url(server.url(""))
+                .build();
+        ResponseBody responseBody = client.newCall(request).execute().body();
+
+        Assert.assertNotNull(responseBody);
+        Assert.assertNotEquals(0, responseBody.contentLength());
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        SimpleEntity response = objectMapper.readValue(responseBody.byteStream()
+                , SimpleEntity.class);
```
Comment (KevinGilmore) [inline libraries-2/src/test/java/com/baeldung/okhttp/ResponseDecoderUnitTest.java]: let's move the comma to the end of the previous line

File: libraries-2/src/test/java/com/baeldung/okhttp/ResponseDecoderUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    MockResponse mockResponse;
+
+    OkHttpClient client;
+
+
+    @Before
+    public void setUp() {
+        sampleResponse = new SimpleEntity("Baeldung");
+        client = new OkHttpClient.Builder()
+                .build();
```
Comment (KevinGilmore) [inline libraries-2/src/test/java/com/baeldung/okhttp/ResponseDecoderUnitTest.java]: indentation on continuation lines should be only two spaces, so let's fix this throughout all the Java files

File: libraries-2/src/test/java/com/baeldung/okhttp/ResponseDecoderUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+                .url(server.url(""))
+                .build();
+        ResponseBody responseBody = client.newCall(request).execute().body();
+
+        Assert.assertNotNull(responseBody);
+        Assert.assertNotEquals(0, responseBody.contentLength());
+
+        Gson gson = new Gson();
+        SimpleEntity response = gson.fromJson(new InputStreamReader(responseBody.byteStream())
+                , SimpleEntity.class);
```
Comment (KevinGilmore) [inline libraries-2/src/test/java/com/baeldung/okhttp/ResponseDecoderUnitTest.java]: move comma up

**Comment by eric-martin** [comment]
[retest this please]
Note: This comment is to trigger the Jenkins CI build

**Pull request merged**